codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/Instantiator.kt

```diff
@@ -186,18 +187,24 @@ class Instantiator(
             true -> ".to_ascii_lowercase()"
             else -> ""
         }
-        if (data.members.isNotEmpty()) {
+        if (data.members.isEmpty()) {
+            writer.write("#T::new()", RustType.HashMap.RuntimeType)
+        } else {
             writer.rustBlock("") {
                 write("let mut ret = #T::new();", RustType.HashMap.RuntimeType)
                 data.members.forEach { (k, v) ->
-                    withBlock("ret.insert(${k.value.dq()}.to_string()$lowercase,", ");") {
+                    withBlock("ret.insert(", ");") {
```

Comment:
you should actually just be able to use `renderMember(this, shape.key, k, ctx)`

Reply:
Great suggestion. Fixed.